libmachine/migrate.go

```diff
@@ -0,0 +1,70 @@
+package libmachine
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/version"
+)
+
+func getMigratedHostMetadata(data []byte) (*HostMetadata, error) {
+	// HostMetadata is for a "first pass" so we can then load the driver
+	var (
+		hostMetadata *HostMetadataV0
+	)
+
+	if err := json.Unmarshal(data, &hostMetadata); err != nil {
+		return &HostMetadata{}, err
+	}
+
+	migratedHostMetadata := MigrateHostMetadataV0ToHostMetadataV1(hostMetadata)
+
+	return migratedHostMetadata, nil
+}
+
+func MigrateHost(h *Host, data []byte) (*Host, error) {
+	var (
+		migratedHostMetadata *HostMetadata
+		err                  error
+	)
+
+	hostV0 := &HostV0{}
+
+	migratedHostMetadata, err = getMigratedHostMetadata(data)
+	if err != nil {
+		return &Host{}, err
+	}
+
+	authOptions := migratedHostMetadata.HostOptions.AuthOptions
+
+	driver, err := drivers.NewDriver(
+		migratedHostMetadata.DriverName,
+		h.Name,
+		h.StorePath,
+		authOptions.CaCertPath,
+		authOptions.PrivateKeyPath,
+	)
+	if err != nil {
+		return &Host{}, err
+	}
+
+	for h.ConfigVersion = migratedHostMetadata.ConfigVersion; h.ConfigVersion < version.ConfigVersion; h.ConfigVersion++ {
+		switch h.ConfigVersion {
+		case 0:
```

Comment:
It may not be immediately obvious, but in the case of future migrations, we will simply move the current `Host` to, say, `HostV1`, add another `case` in this `switch` block here, and add the relevant migration functions.